xbmc/addons/Skin.cpp

```diff
@@ -774,4 +812,17 @@ bool CSkinInfo::SettingsToXML(CXBMCTinyXML &doc) const
   return true;
 }
 
+void CSkinSettingUpdateHandler::OnTimeout()
+{
+  m_addon.SaveSettings();
+}
+
+void CSkinSettingUpdateHandler::TriggerSave()
+{
+  if (m_timer.IsRunning())
+    m_timer.Restart();
```

Comment:
This could result in the settings never being saved if a skin changes some setting at least once every 499ms.